### DISCUSSION TITLE
busy status. I can't get readings. I only get -2

### DISCUSSION THREAD

**Discussion by sondasur** [yasir-shahzad/CS5530#6] [Q&A]
Hello, my name is Joel, I would like to first thank you for your contribution to the world with your great library. I tell you that I have compiled your example, "CS5530_basic", on Arduino Uno, (ver 1.8.19), and I can only print "-2".  I am using the CS5530 shield v1.0 card, which you recommend.
My configuration is:
vcc: 5.0
vdd:5.0
+vref:2.6
-vref: 0
I appreciate your valuable help
Best regards

**Comment by yasir-shahzad** [comment]
Hi Sondasur, I'll get back to you in 5 hours. I'm currently on duty, and I'll be able to assist you in solving this problem after 5hours. Thanks for your patience.

**Reply by xiayu0098** [reply]
I got '-2' as well after running CS5530_basic.ino with Arduino Nano. 

 Can you provide some help ?
 
 Many thanks.

**Reply by yasir-shahzad** [reply]
Hi @xiayu0098 , could you please share the connection diagram of your setup?

**Reply by xiayu0098** [reply]
Thanks for your reply - much appreciated.  The diagram is shown blow. 
3.3V is not just sourced from FT232, it is provided by an external LM1117 - 3.3V.

![longshot20250506200806](https://github.com/user-attachments/assets/3d5e768f-1d98-4cf5-a551-ceef3173f3dd)

The SPI pin connections are as follows:
	•	MOSI → D11
	•	MISO → D12
	•	SCLK → D13
	•	CS (Chip Select) → D10

**Comment by yasir-shahzad** [comment]
I have hardware in my office. I will run the example tomorrow morning and will share the results with you.

**Reply by xiayu0098** [reply]
Any good news about this? Many thanks!
